spring-webflux/src/test/java/org/springframework/web/reactive/function/MultipartIntegrationTests.java

```diff
@@ -95,12 +103,29 @@ void parts(HttpServer httpServer) throws Exception {
 				.verify(Duration.ofSeconds(5));
 	}
 
-	@ParameterizedHttpServerTest
+	static Stream<HttpServer> httpServersWithoutUndertow() {
+		return Stream.of(
+				new JettyHttpServer(),
+				new ReactorHttpServer(),
+				new TomcatHttpServer()
+		);
+	}
+
+	@ParameterizedTest(name = "[{index}] {0}")
+	@MethodSource("httpServersWithoutUndertow")
 	void transferTo(HttpServer httpServer) throws Exception {
 		// TODO: check why Undertow fails
-		if (httpServer instanceof UndertowHttpServer) {
-			return;
-		}
+		//  UndertowHttpServer is unstable, so independently use transferToWithUndertow to test
+		verifyTransferTo(httpServer);
+	}
+
+	@Disabled("Unstable on Undertow,Use @RepeatedTest(100) for verify")
+	@Test
+	void transferToWithUndertow() throws Exception {
```

Comment:
I agree that extracting the test for Undertow to a dedicated method makes it easier to seek a solution for the time being.

But please make sure to shutdown the server at the end of the test method, preferably in a `finally` block.

Reply:
Hi @sbrannen ,
I found the parent class "AbstractHttpHandlerIntegrationTests" use `@AfterEach` to shutdown the server.  Do you sure I need add `finally` block?
Thanks

Reply:
You're right. You're still invoking `startServer(httpServer)`, so `stopServer()` will still get invoked. So there's no need for the extra `finally` block.